tests/test_qasm_examples.py

```diff
@@ -39,12 +59,46 @@ def test_deserialize_and_serialize(fname):
         else:
             formatted_qasm_lines.extend(clean(line) + ";" for line in split_lines if clean(line))
 
+    # moving braces being placed on the same line to the next line (to fit with
+    # what's being returned by the serializer)
+    formatted_qasm_lines = fix_braces(formatted_qasm_lines)
+
     tree = parser.qasm_parser.parse("\n".join(formatted_qasm_lines))
     tree = parser.QASMToIRTransformer().transform(tree)
     serialized_lines = tree.serialize().split("\n")
     serialized_lines = [clean(line) for line in serialized_lines]
     serialized_lines = [line for line in serialized_lines if line]
 
+    # remove all whitespaces temporarily for tests to pass; should come up with
+    # better solution
+    for i, l in enumerate(formatted_qasm_lines):
+        formatted_qasm_lines[i] = l.replace(" ", "")
+        serialized_lines[i] = serialized_lines[i].replace(" ", "")
```

Comment:
This shouldn't be necessary. The whitespace is already normalized in the `clean` function. :thinking:. It might be the `fix_braces` function above? 

What happens if you repeat lines 48-49 here again?

Reply:
Oh, I see. Unfortunately, that still doesn't solve this. The issue lies with the cases where the formatted lines differ like e.g. `if(syn==1) x q[0];`. This string would only be split at the existing whitespaces while the serialized string might look like `'if (syn == 1) x q[0];'.split()` and thus wouldn't split in the same places. Does this make sense?

Reply:
:thinking: hmm, wasn't able to follow